tests/test_ai_chat.py

```
@@ -0,0 +1,314 @@
+                "message": "Complex nutrition analysis with detailed recommendations",
+                "provider": "auto",
+            },
+        )
+
+        assert response.status_code == 200
+        data = response.json()
+        assert data["provider"] == "openai"
+        assert data["estimated_cost"] > 0
+        assert data["complexity"] == "complex"
+
+    def test_cost_estimate_forced_provider(self, client, mock_ai_router):
```

Ruff (0.14.1)

Unused method argument: `mock_ai_router`

(ARG002)